libraries/grpc-sdk/src/classes/ConduitModule.ts

```diff
@@ -1,24 +1,25 @@
 import { getModuleNameInterceptor } from '../interceptors';
 import { CompatServiceDefinition } from 'nice-grpc/lib/service-definitions';
 import { Channel, Client, createChannel, createClientFactory } from 'nice-grpc';
+import { HealthDefinition, HealthCheckResponse } from '../protoUtils/grpc_health_check';
+import { EventEmitter } from 'events';
 
 export class ConduitModule<T extends CompatServiceDefinition> {
   active: boolean = false;
   private _client?: Client<T>;
+  private _healthClient?: Client<typeof HealthDefinition>
   protected channel?: Channel;
   protected protoPath?: string;
   protected type?: T;
-  protected readonly _name: string;
-  protected readonly _url: string;
+  protected readonly _clientName: string;
+  protected readonly _serviceName: string;
+  protected readonly _serviceUrl: string;
+  protected readonly healthCheckEmitter = new EventEmitter();
 
-  // private readonly _interceptors: any[];
-
-  constructor(name: string, url: string) {
-    this._name = name;
-    this._url = url;
-    // this._interceptors = [
-    //   getModuleNameInterceptor(this._name),
-    // ];
+  constructor(clientName: string, serviceName: string, serviceUrl: string) {
+    this._clientName = clientName;
```

Comment:
`clientName` is not `serviceName` for any module as the former refers to the module initializing a gRPC client.
I could rename these to `clientModuleName` and `serverModuleName` for clarification.

Both the check and watch functions accept a service variable, so the latter could just emit an event based on that, but service names are not necessarily unique across modules so that is probably unwise.
Ideally the watcher function should emit an event formatted with both the server module name and the service name.

The watcher function could accept an event name instead, that way we can get rid of it while at the same time make it so that module authors know what event name to listen for.
If we choose not to, we should probably at the very least return the formatted event name so module authors know what to listen for without having to read how `ConduitModule` works.